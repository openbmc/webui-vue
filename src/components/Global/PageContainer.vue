<template>
  <main id="main-content" class="page-container">
    <slot />
  </main>
</template>

<script>
import SetFocusMixin from '@/components/Mixins/SetFocusMixin';
export default {
  name: 'PageContainer',
  mixins: [SetFocusMixin],
  created() {
    this.$root.$on('skip-navigation', () => {
      this.setFocus(this.$el);
    });
  },
};
</script>
<style lang="scss" scoped>
main {
  width: 100%;
  height: 100%;
  padding-top: $spacer * 1.5;
  padding-bottom: $spacer * 3;
  padding-left: $spacer;
  padding-right: $spacer;

  &:focus-visible {
    box-shadow: inset 0 0 0 2px theme-color('primary');
    outline: none;
  }

  @include media-breakpoint-up($responsive-layout-bp) {
    padding-left: $spacer * 2;
  }
}
</style>
