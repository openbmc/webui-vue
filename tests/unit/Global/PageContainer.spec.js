import { mount, createLocalVue } from '@vue/test-utils';
import PageContainer from '@/components/Global/PageContainer';

const localVue = createLocalVue();

describe('PageContainer.vue', () => {
  const wrapper = mount(PageContainer, {
    localVue,
    mocks: {
      $t: (key) => key,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should render main element', () => {
    expect(wrapper.find('main').exists()).toBe(true);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
