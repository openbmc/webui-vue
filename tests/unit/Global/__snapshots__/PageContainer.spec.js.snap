// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer.vue should render correctly 1`] = `
<main
  class="page-container"
  id="main-content"
/>
`;
